pkg/logs/input/docker/launcher.go

```diff
@@ -169,3 +175,45 @@ func (l *Launcher) stopTailer(containerID string) {
 		delete(l.tailers, containerID)
 	}
 }
+
+func (l *Launcher) restartTailer(containerID string) {
+	backoffDuration := 1 * time.Second
+	backoffMax := 60 * time.Second
+	var tailer *Tailer
+	var source *config.LogSource
+
+	for i := 0; ; i++ {
+		if backoffDuration > backoffMax {
+			backoffDuration = backoffMax
+		}
+
+		if i == 0 {
+			oldTailer, _ := l.tailers[containerID]
+			source = oldTailer.source
+			oldTailer.Stop()
+		}
+
+		tailer = NewTailer(l.cli, containerID, source, l.pipelineProvider.NextPipelineChan(), l.erroredContainerID)
+
+		// compute the offset to prevent from missing or duplicating logs
+		since, err := Since(l.registry, tailer.Identifier(), service.Before)
+		if err != nil {
```

Comment:
I understand the motivation behind fetching the offset before resuming but can this ever recover ?
If it fails once I think it is going to fail forever.

Reply:
We should probably use the same logic as other tailers, which I think is to log the error and tail from the end (which is likely what `Since` returns in case of error).

Reply:
👍